### PR TITLE
Rename to extensions

### PR DESCRIPTION
**Pull request by mloufra**
### Description
Change the class (PluginRequest, PluginResponse) to (InitializeExtensionsRequest, InitializeExtensionsResponse), and the variable name related to class (PluginRequest, PluginResponse).

Equivalent PR on OpenSearch : https://github.com/opensearch-project/OpenSearch/pull/3951

### Issues Resolved

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/sdk/ExtensionsRunner.java [hunk 1]
```diff
@@ -20,8 +20,8 @@
 import org.opensearch.common.network.NetworkService;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.common.util.PageCacheRecycler;
-import org.opensearch.discovery.PluginRequest;
-import org.opensearch.discovery.PluginResponse;
+import org.opensearch.discovery.InitializeExtensionsRequest;
+import org.opensearch.discovery.InitializeExtensionsResponse;
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/sdk/ExtensionsRunner.java:24]: The changes in this file look good but you are importing changed class names in the `org.opensearch.discovery` package.  How does this class access those dependencies?  Is there another PR against a different repository that this is linked to?

Comment (mloufra) [reply src/main/java/org/opensearch/sdk/ExtensionsRunner.java:24]: Equivalent PR on OpenSearch : https://github.com/opensearch-project/OpenSearch/pull/3951

**Review by dbwiddis — APPROVED** [review]

File: src/main/java/org/opensearch/sdk/ExtensionsRunner.java [hunk 2]
```diff
@@ -97,12 +97,12 @@ public DiscoveryNode getOpensearchNode() {
      * @param pluginRequest  The request to handle.
```
Comment (peternied) [inline src/main/java/org/opensearch/sdk/ExtensionsRunner.java:97]: super nit: update javadoc

**Comment by owaiskazi19** [comment]
Let's merge this after https://github.com/opensearch-project/OpenSearch/pull/3951 gets merged

**Pull request closed without merging**